rasa_sdk/forms.py

```diff
@@ -656,29 +559,8 @@ async def run(
     def __str__(self) -> Text:
         return f"FormAction('{self.name()}')"
 
-    def _get_entity_type_of_slot_to_fill(
-        self,
-        slot_to_fill: Optional[Text],
-    ) -> Optional[Text]:
-        if not slot_to_fill:
-            return None
-
-        mappings = self.get_mappings_for_slot(slot_to_fill)
-        mappings = [m for m in mappings if m.get("type") == "from_entity"]
-
-        if not mappings:
-            return None
-
-        entity_type = mappings[0].get("entity")
-
-        for i in range(1, len(mappings)):
-            if entity_type != mappings[i].get("entity"):
-                return None
-
-        return entity_type
-
 
-class FormValidationAction(Action, ABC):
+class ValidationAction(Action, ABC):
```

Comment:
I haven't yet looked in detail in all the PR diff, however thought of flagging that this part of the issue is incomplete. This class needs to be also redesigned to work for slots that can be extracted in a custom way outside forms, at the moment it only serves forms. This is what I was referring when asking about the naming convention when the action should set slots outside forms (e.g., the `validate_slots` prefix I used in the rasa default action). Let me know if you prefer to jump on a quick call today/tomorrow to discuss this?

Reply:
Let's jump on a call early tomorrow.